test/unit/validators/hostname_test.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:test/test.dart';
+import 'package:utopia_framework/src/validators/hostname.dart';
+
+void main() {
+  final hostname = Hostname(['abc123', '*.test']);
+
+  group(
+    'HostName |',
+    () {
+      test(
+        'getDescription(): should return proper description message',
+        () {
+          const expectedValue =
+              'Value must be a valid hostname without path, port or protocol.';
+
+          expect(hostname.getDescription(), expectedValue);
+        },
+      );
+
+      test(
+        'getType(): should return proper data type',
+        () {
+          expect(hostname.getType(), 'string');
+        },
+      );
+
+      test(
+        'isArray(): should return false',
+        () {
+          expect(hostname.isArray(), false);
+        },
+      );
+
+      test(
+        'isValid(): should check the validity and return proper boolean value',
+        () {
+          expect(hostname.isValid(1), false);
```

Comment:
Let's add more tests that tests the wildcard and all. 
ref: https://github.com/utopia-php/framework/blob/master/tests/Validator/HostnameTest.php